controller/gif-controller.js

```diff
@@ -0,0 +1,95 @@
+const db = require("../db/db");
+const cloudinary = require("cloudinary");
+require("dotenv").config();
+
+const createGif = async (req, res) => {
+  // get the token details from the authmiddleware
+  const user = req.user;
```

Comment:
```suggestion
  const userId = req.user?.id;
```
Read About [Optional Chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining)